pax-warp-core/src/main/java/org/ops4j/pax/warp/core/update/impl/UpdateSqlGenerator.java

```diff
@@ -169,8 +169,13 @@ public VisitorAction enter(TruncateTable action) {
 
     @Override
     public VisitorAction enter(RunSql action) {
-        String sql = action.getValue();
-        runStatement(sql);
+        String selectedDbms = action.getDbms();
+        String currentSubprotocol = dbms.getSubprotocol();
+
+        if (selectedDbms == null || currentSubprotocol.contains(selectedDbms)) {
```

Comment:
Shouldn't this be `selectedDbms.contains(currentSubprotocol)`?

Anyway, this is not very robust, you'd better split the comma-separated list at ",\s*" delimiters and match the parts.